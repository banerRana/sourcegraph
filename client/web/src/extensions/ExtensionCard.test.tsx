import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router'
import { CompatRouter } from 'react-router-dom-v5-compat'

import { PlatformContext } from '@sourcegraph/shared/src/platform/context'

import { AuthenticatedUser } from '../auth'

import { ExtensionCard } from './ExtensionCard'

const typescriptRawManifest = JSON.stringify({
    activationEvents: ['*'],
    browserslist: [],
    categories: ['Programming languages'],
    contributes: {},
    description: 'TypeScript/JavaScript code graph',
    devDependencies: {},
    extensionID: 'sourcegraph/typescript',
    main: 'dist/extension.js',
    name: 'typescript',
    publisher: 'sourcegraph',
    readme:
        '# Codecov Sourcegraph extension \n\n A [Sourcegraph extension](https://docs.sourcegraph.com/extensions) for showing code coverage information from [Codecov](https://codecov.io) on GitHub, Sourcegraph, and other tools. \n\n ## Features \n\n - Support for GitHub.com and Sourcegraph.com \n - Line coverage overlays on files (with green/yellow/red background colors) \n - Line branches/hits annotations on files \n - File coverage ratio indicator (`Coverage: N%`) and toggle button \n - Support for using a Codecov API token to see coverage for private repositories \n - File and directory coverage decorations on Sourcegraph \n\n ## Usage \n\n ### On GitHub using the Chrome extension \n 1. Install [Sourcegraph for Chrome](https://chrome.google.com/webstore/detail/sourcegraph/dgjhfomjieaadpoljlnidmbgkdffpack) \n 2. [Enable the Codecov extension on Sourcegraph](https://sourcegraph.com/extensions/sourcegraph/codecov) \n 3. Visit [tuf_store.go in theupdateframework/notary on GitHub](https://github.com/theupdateframework/notary/blob/master/server/storage/tuf_store.go) (or any other file in a public repository that has Codecov code coverage) \n 4. Click the `Coverage: N%` button to toggle Codecov test coverage background colors on the file (scroll down if they arenâ€™t immediately visible) \n\n',
    scripts: {},
    tags: [],
    url:
        'https://sourcegraph.com/-/static/extension/7895-sourcegraph-typescript.js?c4ri18f15tv4--sourcegraph-typescript',
    version: '0.0.0-DEVELOPMENT',
})

const registryExtensionNode = {
    id: 'test-extension-1',
    publisher: null,
    extensionID: 'sourcegraph/typescript',
    extensionIDWithoutRegistry: 'sourcegraph/typescript',
    name: 'typescript',
    manifest: {
        raw: typescriptRawManifest,
        description: 'TypeScript/JavaScript code graph',
    },
    createdAt: '2019-01-26T03:39:17Z',
    updatedAt: '2019-01-26T03:39:17Z',
    url: '/extensions/sourcegraph/typescript',
    remoteURL: 'https://sourcegraph.com/extensions/sourcegraph/typescript',
    registryName: 'sourcegraph.com',
    isLocal: false,
    isWorkInProgress: false,
    viewerCanAdminister: false,
}

describe('ExtensionCard', () => {
    const NOOP_PLATFORM_CONTEXT: PlatformContext = {} as any

    const mockUser = {
        id: 'userID',
        username: 'username',
        email: 'user@me.com',
        siteAdmin: true,
    } as AuthenticatedUser

    test('renders', () => {
        expect(
            render(
                <MemoryRouter>
                    <CompatRouter>
                        <ExtensionCard
                            node={{
                                id: 'x/y',
                                manifest: {
                                    activationEvents: ['*'],
                                    description: 'd',
                                    url: 'https://example.com',
                                    icon: 'data:image/png,abcd',
                                },
                                registryExtension: registryExtensionNode,
                            }}
                            subject={{ id: 'u', viewerCanAdminister: false }}
                            viewerSubject={{
                                __typename: 'User',
                                username: 'u',
                                displayName: 'u',
                                id: 'u',
                                viewerCanAdminister: false,
                            }}
                            siteSubject={{
                                __typename: 'Site',
                                id: 's',
                                viewerCanAdminister: true,
                                allowSiteSettingsEdits: true,
                            }}
                            settingsCascade={{ final: null, subjects: null }}
                            platformContext={NOOP_PLATFORM_CONTEXT}
                            enabled={false}
                            enabledForAllUsers={false}
                            isLightTheme={false}
                            settingsURL="/settings/foobar"
                            authenticatedUser={mockUser}
                        />
                    </CompatRouter>
                </MemoryRouter>
            ).asFragment()
        ).toMatchSnapshot()
    })
})
