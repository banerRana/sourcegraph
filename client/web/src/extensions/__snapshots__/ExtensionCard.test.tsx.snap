// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExtensionCard renders 1`] = `
<DocumentFragment>
  <div
    class="position-relative flex-1 extensionCard"
  >
    <div
      class="cardBody p-0 extension-card__body d-flex flex-column position-relative"
    >
      <div
        class="d-flex align-items-center backgroundSection blue"
      >
        <img
          alt=""
          class="icon"
          src="data:image/png,abcd"
        />
      </div>
      <div
        class="w-100 flex-grow-1 detailsSection"
      >
        <div
          class="mb-2"
        >
          <h3
            class="h3 mb-0 mr-1 text-truncate flex-1"
          >
            <a
              class=""
              href="/extensions/x/y"
            >
              typescript
            </a>
          </h3>
          <span>
            by sourcegraph
            <svg
              class="mdi-icon iconInline logo"
              fill="none"
              viewBox="0 -2 22 22"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M4.52023 2.60624L7.99231 15.3189C8.22611 16.173 9.10635 16.6761 9.95913 16.443C10.8119 16.2085 11.3137 15.3263 11.0805 14.4722L7.60779 1.75882C7.374 0.904702 6.49375 0.401608 5.64097 0.635403C4.78886 0.869197 4.28643 1.75145 4.52023 2.60624Z"
                fill="#ff5543"
                fill-rule="evenodd"
              />
              <path
                clip-rule="evenodd"
                d="M10.9514 2.55785L2.25612 12.3913C1.66996 13.0545 1.73092 14.0687 2.39278 14.6562C3.05464 15.2437 4.06619 15.1821 4.65235 14.5189L13.3476 4.68544C13.9338 4.02224 13.8728 3.00869 13.211 2.42119C12.5491 1.83368 11.5382 1.89465 10.9514 2.55785Z"
                fill="#a112ff"
                fill-rule="evenodd"
              />
              <path
                clip-rule="evenodd"
                d="M1.09849 8.00782L13.499 12.1156C14.3384 12.3943 15.2434 11.9374 15.5214 11.0967C15.7994 10.2553 15.3432 9.34829 14.5045 9.07095L2.10333 4.96112C1.26395 4.68378 0.358918 5.13931 0.0815799 5.98071C-0.196428 6.8221 0.259103 7.72914 1.09849 8.00782Z"
                fill="#00cbec"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </div>
        <div
          class="mt-3 description"
        >
          <span
            class=""
          >
            d
          </span>
        </div>
      </div>
      <div
        class="d-flex flex-column align-items-end mt-1 togglesSection"
      >
        <div
          class="px-1"
        >
          <span
            class="text-muted small"
          >
            Disabled
          </span>
        </div>
        <div
          class="px-1 mt-2"
        >
          <span
            class="text-muted"
          >
            Disabled by default for all
          </span>
          <button
            aria-checked="false"
            class="toggle mx-2 inlineCenter"
            data-testid="extension-toggle-x/y"
            role="switch"
            title="Click to enable"
            type="button"
            value="0"
          >
            <span
              class="bar"
            />
            <span
              class="knob"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
